arithmetic-engine/src/main/kotlin/com/raccoon/backend/arithmetic/internal/SimpleParser.kt

```diff
@@ -0,0 +1,133 @@
+package com.raccoon.backend.arithmetic.internal
+
+import com.raccoon.backend.arithmetic.UnexpectedTokenException
+
+/**
+ * A simple arithmetic parser.
+ * Parses a given string and evaluates the expression it contains.
+ *
+ *  Grammar:
+ *  <Input>   : <Expr>EOF
+ *  <Expr>    : <Term><Terms>
+ *  <Terms>   : +<Term><Terms> | -<Term><Terms> | empty
+ *  <Term>    : <Factor><Factors>
+ *  <Factors> : *<Factor><Factors> | /<Factor><Factors> | empty
+ *  <Factor>  : constant | (<Expr>) | -<Factor> | +<Factor>
+ */
+class SimpleParser(
+    private val lexerFactory: LexerFactory = SimpleLexer.factory,
+    private val stackMachine: StackMachine = SimpleStackMachine()
+) {
+    private lateinit var lexer: Lexer
+    private lateinit var input: String
+
+    private fun error(): Nothing = throw UnexpectedTokenException(input, lexer.lastToken.offset)
+    private fun next() = lexer.next()
```

Comment:
I don't think that this is an entity blending because the method isn't available from outside. In fact it's just a shortcut for lexer's call. But here is another problem: we have this shortcut for the `next` method but have no shortcuts for  the `lastToken` property. It makes the code less readable. So I think we need either remove this shortcut or add a shortcut for the `lastToken` property.

Reply:
Fixed.